### DISCUSSION TITLE
Is there a way to persist node_modules?

### DISCUSSION THREAD

**Discussion by ghost** [mrgrain/cdk-esbuild#553] [Q&A]
Hi, thanks for this package, it's really helpful. One question though - is there any way to persist all node_module dependencies or 
select just a few? I'm using puppeteer-core and a chromium binary within a layer and would like to keep those within the node_modules directory rather than be bundled together with source code. 

Thanks!

**Comment by mrgrain** [comment]
Are you looking for [external](https://constructs.dev/packages/@mrgrain/cdk-esbuild/v/4.1.5/api/BuildOptions?lang=typescript#property.external)?

**Comment by hugomallet** [comment]
Hi @mrgrain , it looks external modules are not copied to output

**Reply by mrgrain** [reply]
They are not. Unfortunately it's not what esbuild does.

You should be able to implement this with `copyDir` though.

**Reply by hugomallet** [reply]
I tried but copyDir is executed from local file system I think ? Because I got some platform related issues

**Reply by mrgrain** [reply]
It is. But where else would you like to copy external modules from?

**Reply by hugomallet** [reply]
Isn't it executed from inside a container based on a aws image ?

**Reply by mrgrain** [reply]
No. This package makes a point of running everything on the local machine. 😅 
Might not be super obvious from the current Readme.

**Reply by mrgrain** [reply]
Can I ask about some more background on what you are trying to do?

**Reply by hugomallet** [reply]
Actually I tried this package because of this response : https://github.com/aws/aws-cdk/issues/14215#issuecomment-1246325012
I am desperately trying to build a node lambda edge using cdk v2

**Reply by mrgrain** [reply]
What's the package that you need a different architecture for?

**Reply by hugomallet** [reply]
sharp (which is a native module)

**Reply by mrgrain** [reply]
Ah yes. Image processing. Personally I tend to put these kind of libraries into a layer, instead the code bundle. But that's because I like my bundles small-ish and platform independent. But it just moves the issue to somewhere else. And of course layers are not support by Lambda@Edge.

**Reply by mrgrain** [reply]
Right now, with this package your best solution will be to run the whole thing on the target platform, e.g. from inside a Docker container.

**Reply by hugomallet** [reply]
Ok thanks

**Reply by mrgrain** [reply]
I also found this on the sharp docs:
https://sharp.pixelplumbing.com/install#cross-platform

Maybe you can tweak the install commands you run to make it work.

**Reply by mrgrain** [reply]
Also: https://sharp.pixelplumbing.com/install#aws-lambda
